agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/ResourceParser.java

```diff
@@ -23,18 +23,27 @@
 
 import com.azure.monitor.opentelemetry.exporter.implementation.builders.AbstractTelemetryBuilder;
 import com.azure.monitor.opentelemetry.exporter.implementation.models.ContextTagKeys;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.Strings;
 import io.opentelemetry.sdk.resources.Resource;
 import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
 import java.util.Map;
 
 final class ResourceParser {
 
+  private static final String DEFAULT_SERVICE_NAME = "unknown_service:java";
+
   static void updateRoleNameAndInstance(AbstractTelemetryBuilder builder, Resource resource) {
     Map<String, String> existingTags = builder.build().getTags();
     if (existingTags == null
         || !existingTags.containsKey(ContextTagKeys.AI_CLOUD_ROLE.toString())) {
       String serviceName = resource.getAttribute(ResourceAttributes.SERVICE_NAME);
       String serviceNamespace = resource.getAttribute(ResourceAttributes.SERVICE_NAMESPACE);
+      if (serviceName == null || DEFAULT_SERVICE_NAME.equals(serviceName)) {
+        String websiteSiteName = Strings.trimAndEmptyToNull(System.getenv("WEBSITE_SITE_NAME"));
+        if (websiteSiteName != null) {
+          serviceName = websiteSiteName;
+        }
+      }
```

Comment:
swapping order here would help my reading comprehension 😄
```suggestion
      if (serviceName == null || DEFAULT_SERVICE_NAME.equals(serviceName)) {
        String websiteSiteName = Strings.trimAndEmptyToNull(System.getenv("WEBSITE_SITE_NAME"));
        if (websiteSiteName != null) {
          serviceName = websiteSiteName;
        }
      }
      String serviceNamespace = resource.getAttribute(ResourceAttributes.SERVICE_NAMESPACE);
```